pkg/cmd/server/api/validation/oauth.go

```diff
@@ -96,18 +96,41 @@ func ValidateIdentityProvider(identityProvider api.IdentityProvider) fielderrors
 		case (*api.HTPasswdPasswordIdentityProvider):
 			allErrs = append(allErrs, ValidateFile(provider.File, "provider.file")...)
 
-		case (*api.OAuthRedirectingIdentityProvider):
-			if len(provider.ClientID) == 0 {
-				allErrs = append(allErrs, fielderrors.NewFieldRequired("provider.clientID"))
+		case (*api.GitHubIdentityProvider):
+			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+
+		case (*api.GoogleIdentityProvider):
+			allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
+
+		case (*api.OpenIDIdentityProvider):
```

Comment:
A fully specified config file should be fully specified.  Saying that our file is fully specified except for these N fields that are filled in with default values takes us backwards and it makes it impossible to distinguish between "doesn't have this value" and "use your default".